components/resourceDetails/resourceSharing/policyHeader/index.jsx

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable react/jsx-one-expression-per-line */
+
+import React from "react";
+import PropTypes from "prop-types";
+import { useBem } from "@solid/lit-prism-patterns";
+import clsx from "clsx";
+import { createStyles } from "@material-ui/core";
+import { makeStyles } from "@material-ui/styles";
+
+import styles from "./styles";
+
+const useStyles = makeStyles((theme) => createStyles(styles(theme)));
+
+export default function PolicyHeader({ children, type, isPolicyList }) {
+  const editorsDescription = (
+    <p>
+      <b>Can </b>
+      view, edit and delete this resource
+    </p>
+  );
+
+  const viewersDescription = (
+    <p>
+      <b>Can </b>
+      view but
+      <b> cannot </b>
+      edit or delete this resource
+    </p>
+  );
+
+  const blockedDescription = (
+    <p>
+      <b>Cannot </b>
+      view this resource
+    </p>
+  );
+
+  const PERMISSIONS_TYPE_MAP = {
+    editors: {
```

Comment:
As I suggested earlier, let's aggregate all of the properties used throughout the various component into a super object that you can pass down the components instead of referring to them with a string.

Reply:
The thing is this is the components that uses the most of these, which is why I put this map here. There is another one that uses only an emptyStateText depending on type, but those components also need the type to do other things, like fetching permissions. I didn't want to pass unnecessary props into components.